src/expr/src/scalar/func.rs

```diff
@@ -6932,6 +6931,95 @@ fn make_mz_acl_item<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
     }))
 }
 
+fn array_fill<'a>(
+    datums: &[Datum<'a>],
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    const MAX_SIZE: usize = 1 << 28 - 1;
+    const NULL_ERR: &str = "dimension array or low bound array";
```

Comment:
This reads weird to me as "low bound" instead of "lower bound". Intentional?

Reply:
Straight from the mouth of Tom Lane:

```
postgres=# select array_fill(1, null);
ERROR:  dimension array or low bound array cannot be null
```